cmd/skopeo/proxy.go

```diff
@@ -197,6 +203,34 @@ func (h *proxyHandler) Initialize(args []interface{}) (replyBuf, error) {
 // OpenImage accepts a string image reference i.e. TRANSPORT:REF - like `skopeo copy`.
 // The return value is an opaque integer handle.
 func (h *proxyHandler) OpenImage(args []interface{}) (replyBuf, error) {
+	return h.openImageImpl(args, false)
+}
+
+// isManifestUnknownError is a copy of code from containers/image,
+// please update there first.
+func isManifestUnknownError(err error) bool {
+	var errs dockerdistributionerrcode.Errors
+	if !errors.As(err, &errs) || len(errs) == 0 {
+		return false
+	}
+	err = errs[0]
+	ec, ok := err.(dockerdistributionerrcode.ErrorCoder)
+	if !ok {
+		return false
+	}
+	return ec.ErrorCode() == dockerdistributionapi.ErrorCodeManifestUnknown
+}
+
+// isNotFoundImageError heuristically attempts to determine whether an error
+// is saying the remote source couldn't find the image (as opposed to an
+// authentication error, an I/O error etc.)
+// TODO drive this into containers/image properly
+func isNotFoundImageError(err error) bool {
+	return isManifestUnknownError(err) ||
+		strings.Contains(err.Error(), ": no descriptor found for reference")
```

Comment:
This really should use a Go type created in c/image. Something like https://github.com/containers/image/pull/1675 ? (Warning: Absolutely untested.)

Reply:
https://github.com/containers/image/pull/1675 is now merged in c/image, and vendored into Skopeo.

Reply:
OK, updated to use that, thanks!

Reply:
Thanks.

Tests are now failing, that’s because https://github.com/containers/image/pull/1299 (as mentioned earlier) changes what `isDockerManifestUnknownError` needs to be looking for; I’m afraid I forgot about that part. Just update the code to match the `c/image/docker/docker_client.go` variant, please.